dashboard/api_edx_cache.py

```diff
@@ -16,6 +16,45 @@
 
 
 class CachedEdxUserData:
+    """Represents all edx data related to a User"""
+    # pylint: disable=too-many-instance-attributes
+
+    enrollments = None
+    certificates = None
+    current_grades = None
+    raw_enrollments = None
+    raw_certificates = None
+    raw_current_grades = None
+
+    def __init__(self, user, program=None, include_raw_data=False):
+        """
+        Fetches the given User's edx data and sets object properties
+
+        Args:
+            user (User): a User object
+            program (Program): an optional Program to filter on
+            include_raw_data (bool): Set to True if raw edx data is needed (in addition to actual edX objects)
+        """
+        self.user = user
+        self.program = program
+        self.fetch_and_set_edx_data()
+        if include_raw_data:
+            self.fetch_and_set_raw_data()
+
+    def fetch_and_set_edx_data(self):
+        """Fetches edx data and sets object properties"""
+        self.enrollments = models.CachedEnrollment.get_edx_data(self.user, program=self.program)
+        self.certificates = models.CachedCertificate.get_edx_data(self.user, program=self.program)
+        self.current_grades = models.CachedCurrentGrade.get_edx_data(self.user, program=self.program)
+
+    def fetch_and_set_raw_data(self):
+        """Fetches raw cached data and sets object properties"""
+        self.raw_enrollments = models.CachedEnrollment.get_raw_data(self.user, program=self.program)
+        self.raw_certificates = models.CachedCertificate.get_raw_data(self.user, program=self.program)
+        self.raw_current_grades = models.CachedCurrentGrade.get_raw_data(self.user, program=self.program)
```

Comment:
the query (`data_qset`) is already cached so i figured there's no penalty, but i realize that could potentially change without us noticing it. the objects above are edX objects (`edx_api.enrollments.models.Enrollments`, etc). given those objects, is there a reliable/consistent way to get the raw data as it is stored in our own models?